docs/source/nodes_and_pipelines/namespaces.md

```diff
@@ -0,0 +1,336 @@
+
+Since the node names are unique now, we can run the project with:
+
+```python
+kedro run
+```
+
+Logs show that `data_science` and `data_science_2` pipelines were executed successfully with different R2 results. Now, we can see how Kedro-viz renders namespaced pipelines in collapsible "super nodes":
+
+```python
+kedro viz run
+```
+
+After running viz, we can see two equal pipelines: `data_science` and `data_science_2`:
+
+![namespaces uncollapsed](../meta/images/namespaces_uncollapsed.png)
+
+We can collapse all namespaced pipelines (in our case, it's only `data_science_2`) with a special button and see that the `data_science_2` pipeline was collapsed into one super node called `Ds 2`:
+
+![namespaces collapsed](../meta/images/namespaces_collapsed.png)
+
+> Tip: You can use `kedro run --namespace = namespace_name` to run only the specific namespace
+
+
+### How to namespace all pipelines in a project
+
+If we want to make all pipelines in this example fully namespaced, we should:
+
+Modify the `data_processing` pipeline by adding to the `pipeline()` creation function in `src/project_name/pipelines/data_processing/pipeline.py` with the following code:
+```python
+        namespace="data_processing",
+        inputs={"companies", "shuttles", "reviews"},  # Inputs remain the same, without namespace prefix
+        outputs={"model_input_table"},  # Outputs remain the same, without namespace prefix
+```
+Modify the `data_science` pipeline by adding namespace and inputs in the same way as it was done in `data_science_2` pipeline:
+
+```python
+def create_pipeline(**kwargs) -> Pipeline:
+    return pipeline(
+        [base_data_science],
+        parameters={"params:model_options": "params:model_options_1"},
+        namespace="ds_1",
+        inputs={"model_input_table"},
+    )
+```
+
+After executing the pipeline with `kedro run`, the visualisation with `kedro viz run` after collapsing will look like this:
+
+![namespaces collapsed all](../meta/images/namespaces_collapsed_all.png)
+
+## Providing pipeline specific dependencies
+
+* A pipeline **might** have external dependencies specified in a local `requirements.txt` file.
+* Pipeline specific dependencies are scooped up during the [micro-packaging](micro_packaging.md) process.
+* These dependencies need to be manually installed using `pip`:
+```bash
+pip install -r requirements.txt
+```
+
+## Example: Combining disconnected pipelines
+
+Sometimes two pipelines must be connected, but do not share any catalog dependencies. In this example, there is a `lunch_pipeline`, which makes us lunch. The 'verbs', `defrost` and `eat`, are Python functions and the inputs/outputs are food at different points of the process (`frozen`, `thawed` and `food`).
```

Comment:
`thawed` is not present in the example at all

Reply:
Decided to remove the cooking examples.